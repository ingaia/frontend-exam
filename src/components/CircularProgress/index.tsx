import React from 'react';
import CircularProgress from '@material-ui/core/CircularProgress';
import { useStyles } from '../../styles/CircularProgress';

const Component = () => (
  <CircularProgress classes={useStyles()} size={80} thickness={2} />
);

export default Component;
