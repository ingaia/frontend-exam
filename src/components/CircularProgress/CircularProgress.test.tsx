import React from 'react';
import { render } from '@testing-library/react';
import CircularProgress from '.';

describe('<CircularProgress />', () => {
  it('should render correctly', () => {
    const { container } = render(<CircularProgress />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the correct size', () => {
    const { container } = render(<CircularProgress />)

    expect(container.firstChild).toHaveStyle({ 'width': '80px', 'height': '80px' });
  })
})
