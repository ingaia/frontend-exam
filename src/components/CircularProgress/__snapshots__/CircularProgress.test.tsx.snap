// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CircularProgress /> should render correctly 1`] = `
<div
  class="MuiCircularProgress-root makeStyles-root-1 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
  role="progressbar"
  style="width: 80px; height: 80px;"
>
  <svg
    class="MuiCircularProgress-svg"
    viewBox="22 22 44 44"
  >
    <circle
      class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
      cx="44"
      cy="44"
      fill="none"
      r="21"
      stroke-width="2"
    />
  </svg>
</div>
`;
